Adding a new TranslatorV3 connector and deleting DocumentTranslator connector

HamshavathiM opened the pull request:
Adding a new TranslatorV3 connector and deleting DocumentTranslator connector

HamshavathiM commented:
@microsoft-github-policy-service agree company="Microsoft"

From: microsoft-github-policy-service[bot] ***@***.***>
Sent: Monday, February 13, 2023 8:16 PM
To: microsoft/PowerPlatformConnectors ***@***.***>
Cc: Hamshavathi Munibyraiah (Murphy & Associates Inc) ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/PowerPlatformConnectors] Adding a new TranslatorV3 connector and deleting DocumentTranslator connector (PR #2389)


@HamshavathiM<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FHamshavathiM&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7Cf471ed3034594733e60d08db0e423d94%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638119449893460692%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6sPsoYrg4P2YJHRJGshlP%2BxpOHlUW2E0Rdq4tmmAWW4%3D&reserved=0> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@microsoft-github-policy-service agree [company="{your company}"]



Options:

  *   (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

@microsoft-github-policy-service agree



  *   (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term "You" includes me and my employer.

@microsoft-github-policy-service agree company="Microsoft"


Contributor License Agreement
Contribution License Agreement

This Contribution License Agreement ("Agreement") is agreed to by the party signing below ("You"),
and conveys certain license rights to Microsoft Corporation and its affiliates ("Microsoft") for Your
contributions to Microsoft open source projects. This Agreement is effective as of the latest signature
date below.

  1.  Definitions.
"Code" means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to Microsoft under this Agreement.
"Project" means any of the projects owned or managed by Microsoft and offered under a license
approved by the Open Source Initiative (www.opensource.org<https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.opensource.org%2F&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7Cf471ed3034594733e60d08db0e423d94%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638119449893460692%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=eySOVnBEqccUDGJsVtnlsTNrwcvSkOEyLeF02nOLvWs%3D&reserved=0>).
"Submit" is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as "Not a Submission."
"Submission" means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.
  2.  Your Submission. You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.
  3.  Originality of Work. You represent that each of Your Submissions is entirely Your original work.
Should You wish to Submit materials that are not Your original work, You may Submit them separately
to the Project if You (a) retain all copyright and license information that was in the materials as You
received them, (b) in the description accompanying Your Submission, include the phrase "Submission
containing materials of a third party:" followed by the names of the third party and any licenses or other
restrictions of which You are aware, and (c) follow any other instructions in the Project's written
guidelines concerning Submissions.
  4.  Your Employer. References to "employer" in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term "You" in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.
  5.  Licenses.

  *   Copyright License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
parties.
  *   Patent License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.
  *   Other Rights Reserved. Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

  1.  Representations and Warranties. You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
  2.  Notice to Microsoft. You agree to notify Microsoft in writing of any facts or circumstances of which
You later become aware that would make Your representations in this Agreement inaccurate in any
respect.
  3.  Information about Submissions. You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.
  4.  Governing Law/Jurisdiction. This Agreement is governed by the laws of the State of Washington, and
the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
defenses of lack of personal jurisdiction and forum non-conveniens.
  5.  Entire Agreement/Assignment. This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

-
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FPowerPlatformConnectors%2Fpull%2F2389%23issuecomment-1429091870&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7Cf471ed3034594733e60d08db0e423d94%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638119449893460692%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=c4GEq%2B9cO8tbPpuNZe%2Bgb13iJI5sl1aih95BkKlX5Sg%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAYF4VDRKEMGC5QS4LFSPQB3WXMBJVANCNFSM6AAAAAAU3CC5KU&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7Cf471ed3034594733e60d08db0e423d94%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638119449893460692%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=He65z8C78DVJHA7n3Nelx%2Bdz%2FwFrB2K4FLLEEtpLtKk%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

Amjed-Ayoub commented on certified-connectors/TranslatorV3/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,1870 @@
+            }
+          }
+        ],
+        "externalDocs": {
```
@HamshavathiM the "externalDocs" is not required, can you please remove it from all the swagger file?

HamshavathiM commented:
1.  Added "stackOwner" property.
  2.  We want the connector version to be 3.0. As we already have different versions.
  3.  The hostURL is just a place holder, we do not really use it. The host url is constructed from the connection parameters.
  4.  Removed "externalDocs"

From: Amjed Ayoub ***@***.***>
Sent: Tuesday, February 14, 2023 11:59 AM
To: microsoft/PowerPlatformConnectors ***@***.***>
Cc: Hamshavathi Munibyraiah (Murphy & Associates Inc) ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/PowerPlatformConnectors] Adding a new TranslatorV3 connector and deleting DocumentTranslator connector (PR #2389)


@Amjed-Ayoub requested changes on this pull request.

________________________________

In certified-connectors/TranslatorV3/apiProperties.json<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FPowerPlatformConnectors%2Fpull%2F2389%23discussion_r1106227731&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015302953508%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9WXHbYrN6beXqfgTXmVG2CgqpsSea8CDe4Dy%2FtkkNzw%3D&reserved=0>:

> +            "Transliterate",

+            "StartDocumentTranslation",

+            "GetTranslationStatus",

+            "CancelTranslation",

+            "GetTranslationsStatus",

+            "GetDocumentStatus",

+            "GetSupportedDocumentFormats",

+            "GetSupportedGlossaryFormats",

+            "GetSupportedStorageSources",

+            "GetDocumentsStatus",

+            "GetGlossaryFormatVersion"

+          ]

+        }

+      }

+    ],

+    "publisher": "Hamshavathi Munibyraiah (Murphy & Associates Inc)"

@HamshavathiM<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FHamshavathiM&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015302953508%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=1xthXWRBSklO%2BYxCdrGmvcBAMs%2B6N2msfP82qw3V6Ws%3D&reserved=0> please add the "stackOwner" property.

"stackOwner": "Name of the owner of the service or product."

________________________________

In certified-connectors/TranslatorV3/apiDefinition.swagger.json<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FPowerPlatformConnectors%2Fpull%2F2389%23discussion_r1106228784&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015302953508%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=QbPmm1z%2FYc%2FBBhCHI29p4zjuKsGYJHHV0REi7GCHuAo%3D&reserved=0>:

> @@ -0,0 +1,1870 @@

+{

+  "swagger": "2.0",

+  "info": {

+    "title": "Microsoft Translator V3",

+    "description": "Microsoft Translator lets you translate text across multiple languages.",

+    "version": "3.0",

@HamshavathiM<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FHamshavathiM&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015302953508%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=1xthXWRBSklO%2BYxCdrGmvcBAMs%2B6N2msfP82qw3V6Ws%3D&reserved=0> since this is a new connector, can you please change the version to "1.0"?

________________________________

In certified-connectors/TranslatorV3/apiDefinition.swagger.json<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FPowerPlatformConnectors%2Fpull%2F2389%23discussion_r1106229839&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015303110259%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=42RLb6CdvT9Bm%2FsJD9TBXfaSf9fKyte%2B674iqe82Yz0%3D&reserved=0>:

> @@ -0,0 +1,1870 @@

+{

+  "swagger": "2.0",

+  "info": {

+    "title": "Microsoft Translator V3",

+    "description": "Microsoft Translator lets you translate text across multiple languages.",

+    "version": "3.0",

+    "contact": {

+      "name": "Microsoft Translator",

+      "url": "https://www.microsoft.com/translator/",

+      "email": ***@***.******@***.***>"

+    }

+  },

+  "host": "translator-batch-test.cognitiveservices.azure.com",

@HamshavathiM<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FHamshavathiM&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015303110259%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=FM8nfhyVsMwwBrNT91EVCoLOYgqras%2FXyHKDyQo52ys%3D&reserved=0> the host URL should be in production mode and not in testing or staging mode.
Can you please confirm?

________________________________

In certified-connectors/TranslatorV3/apiDefinition.swagger.json<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FPowerPlatformConnectors%2Fpull%2F2389%23discussion_r1106307431&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015303110259%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=BjVnO0J2plribM1bEszThW3D%2BIb4QlTL%2BEmB%2FRPqnK8%3D&reserved=0>:

> +                    "description": "Text to translate.",

+                    "x-ms-visibility": "important",

+                    "default": "Hello"

+                  }

+                },

+                "required": [

+                  "Text"

+                ]

+              },

+              "required": [

+                "items"

+              ]

+            }

+          }

+        ],

+        "externalDocs": {

@HamshavathiM<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FHamshavathiM&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015303110259%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=FM8nfhyVsMwwBrNT91EVCoLOYgqras%2FXyHKDyQo52ys%3D&reserved=0> the "externalDocs" is not required, can you please remove it from all the swagger file?

-
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FPowerPlatformConnectors%2Fpull%2F2389%23pullrequestreview-1298201203&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015303110259%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=j6ugDpB3uDFxUuvThrrVpJZ%2B5VTaFXYMhXut2qu7LNI%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAYF4VDRNYENGZXRCNAXYGTLWXPPXPANCNFSM6AAAAAAU3CC5KU&data=05%7C01%7Cv-hmunibyrai%40microsoft.com%7C3b474884eaf4411f19fd08db0ec5e285%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638120015303110259%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=lb9xUJIs8dB7vM7zwE1Sbh7RdtjjOlX%2Bxjp1gOip0XA%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

Amjed-Ayoub commented:
Hello @HamshavathiM,
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector. 
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.
